### PR TITLE
Update The Demo To Use Composer to Pull Core

### LINKED ISSUE #77: Update Config To Not Commit Wordpress Core
The best practice has become similar to Drupal where core is not committed to your personal repo. Since we need to update this demo anyway to update our docs, this ticket will explore getting everything up to date.

### PR DESCRIPTION
**Pull request by KeyboardCowboy**
Resolves #77.

### CODE REVIEW

**Comment by tugbotdemo** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr78-lchrwu70qd8t3mnmujq2pe3k4nws0e6s.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64946fa152f610cb5746333d

Visual Diffs:
- https://dashboard.tugboatqa.com/64946fa152f610cb5746333d/visualdiffs

**Comment by tugbotdemo** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr78-lchrwu70qd8t3mnmujq2pe3k4nws0e6s.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64946fa152f610cb5746333d

Visual Diffs:
- https://dashboard.tugboatqa.com/64946fa152f610cb5746333d/visualdiffs

**Comment by tugbotdemo** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr78-lchrwu70qd8t3mnmujq2pe3k4nws0e6s.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64946fa152f610cb5746333d

Visual Diffs:
- https://dashboard.tugboatqa.com/64946fa152f610cb5746333d/visualdiffs

**Comment by tugbotdemo** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr78-lchrwu70qd8t3mnmujq2pe3k4nws0e6s.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64946fa152f610cb5746333d

Visual Diffs:
- https://dashboard.tugboatqa.com/64946fa152f610cb5746333d/visualdiffs

**Comment by tugbotdemo** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr78-lchrwu70qd8t3mnmujq2pe3k4nws0e6s.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64946fa152f610cb5746333d

Visual Diffs:
- https://dashboard.tugboatqa.com/64946fa152f610cb5746333d/visualdiffs

**Comment by KeyboardCowboy** [comment]
Looks like the demo is broken due to the paths not properly mapping to the docroot subdirectory.  Lots of 404s on assets and stylesheets.

**Comment by tugbotdemo** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr78-lchrwu70qd8t3mnmujq2pe3k4nws0e6s.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64946fa152f610cb5746333d

Visual Diffs:
- https://dashboard.tugboatqa.com/64946fa152f610cb5746333d/visualdiffs

**Comment by tugbotdemo** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr78-lchrwu70qd8t3mnmujq2pe3k4nws0e6s.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64946fa152f610cb5746333d

Visual Diffs:
- https://dashboard.tugboatqa.com/64946fa152f610cb5746333d/visualdiffs

**Comment by tugbotdemo** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr78-vvg1syntykx8rfmnpy8dpo6hgjxvxiic.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64ac795865d7012a0bb7f888

**Comment by tugbotdemo** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr78-vvg1syntykx8rfmnpy8dpo6hgjxvxiic.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64ac795865d7012a0bb7f888

**Comment by KeyboardCowboy** [comment]
I'm not sure why this is still happening, but the preview's links in the head are pointing to the base preview.  I removed the lines of code in the build phase that seem to rewrite those paths, but it's still happening.  Is this locked into the database dump somehow?

**Comment by tugbotdemo** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr78-vvg1syntykx8rfmnpy8dpo6hgjxvxiic.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64ac795865d7012a0bb7f888

**Comment by tugbotdemo** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr78-vvg1syntykx8rfmnpy8dpo6hgjxvxiic.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64ac795865d7012a0bb7f888

**Comment by tugbotdemo** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr78-vvg1syntykx8rfmnpy8dpo6hgjxvxiic.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64ac795865d7012a0bb7f888

**Comment by tugbotdemo** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr78-vvg1syntykx8rfmnpy8dpo6hgjxvxiic.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64ac795865d7012a0bb7f888

**Comment by KeyboardCowboy** [comment]
I figured it out.  It's fixed.  Adjusted a couple of lines in the `build` phase to replace the outdated URL with the correct preview path.

**Comment by tugbotdemo** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr78-vvg1syntykx8rfmnpy8dpo6hgjxvxiic.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64ac795865d7012a0bb7f888

**Comment by tugbotdemo** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr78-vvg1syntykx8rfmnpy8dpo6hgjxvxiic.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64ac795865d7012a0bb7f888

**Comment by tugbotdemo** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr78-vvg1syntykx8rfmnpy8dpo6hgjxvxiic.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64ac795865d7012a0bb7f888

**Comment by tugbotdemo** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr78-vvg1syntykx8rfmnpy8dpo6hgjxvxiic.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64ac795865d7012a0bb7f888

**Comment by tugbotdemo** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr78-vvg1syntykx8rfmnpy8dpo6hgjxvxiic.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64ac795865d7012a0bb7f888

File: .tugboat/config.yml [hunk 1]
```diff
@@ -1,55 +1,95 @@
...
+            # so we want to rewrite the base preview's URL to this preview's.  However, if this preview does NOT have a
+            # base preview, then the original URL from the database export will need to be rewritten.
             build: |
                 if [ "x${TUGBOAT_BASE_PREVIEW}" != "x" ]; then
                     wp --allow-root --path="${DOCROOT}" search-replace "${TUGBOAT_BASE_PREVIEW_URL_HOST}" "${TUGBOAT_SERVICE_URL_HOST}" --skip-columns=guid
+                    echo "Replacing ${TUGBOAT_BASE_PREVIEW_URL_HOST} with ${TUGBOAT_SERVICE_URL_HOST}"
                 else
-                    wp --allow-root --path="${DOCROOT}" search-replace 'main-hbpfbyk632i25rgdo9caeoo9oxs4mrtd.tugboat.qa' "${TUGBOAT_SERVICE_URL_HOST}" --skip-columns=guid
+                    wp --allow-root --path="${DOCROOT}" search-replace 'pr78-vvg1syntykx8rfmnpy8dpo6hgjxvxiic.tugboatqa.com' "${TUGBOAT_SERVICE_URL_HOST}" --skip-columns=guid
+                    echo "Replacing pr78-vvg1syntykx8rfmnpy8dpo6hgjxvxiic.tugboatqa.com with ${TUGBOAT_SERVICE_URL_HOST}"
```
Comment (q0rban) [inline .tugboat/config.yml:88]: Has the database been modified in this PR? If not, these lines should not change.

File: .tugboat/config.yml [hunk 2]
```diff
@@ -1,55 +1,95 @@
...
                 fi
 
             clone:
-              -  wp --allow-root --path="${DOCROOT}" search-replace "${TUGBOAT_BASE_PREVIEW_URL_HOST}" "${TUGBOAT_SERVICE_URL_HOST}" --skip-columns=guid
+            -  wp --allow-root --path="${DOCROOT}" search-replace "${TUGBOAT_BASE_PREVIEW_URL_HOST}" "${TUGBOAT_SERVICE_URL_HOST}" --skip-columns=guid
```
Comment (q0rban) [inline .tugboat/config.yml:92]: Looks like some white space might have gotten messed up here.

File: .tugboat/config.yml [hunk 3]
```diff
@@ -60,8 +100,10 @@ services:
         visualdiff:
             threshold: 90
 
+    # What to call the service hosting MySQL. This name also acts as the
+    # hostname to access the service by from the php service.
     mysql:
-        image: tugboatqa/mysql:5
+        image: tugboatqa/mysql:5-debian
```
Comment (q0rban) [inline .tugboat/config.yml:106]: 👏

**Review by q0rban — COMMENTED** [review]
Nice work, Chris!

File: .tugboat/config.yml (hunk 1 above)
Comment (KeyboardCowboy) [reply .tugboat/config.yml:88]: Yes, I used the preview to dump a new database with "The Tug Life" content removed and any necessary database updates run that I could see.

**Comment by tugbotdemo** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr78-vvg1syntykx8rfmnpy8dpo6hgjxvxiic.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64ac795865d7012a0bb7f888

**Review by q0rban — APPROVED** [review]

**Comment by tugbotdemo** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr78-vvg1syntykx8rfmnpy8dpo6hgjxvxiic.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64ac795865d7012a0bb7f888

**Pull request merged**